src/Pages/ContentListTable/components/SnapshotPackageModal/SnapshotPackageModal.tsx

```diff
@@ -0,0 +1,253 @@
+    color: global_Color_200.value,
+  },
+  mainContainer: {
+    backgroundColor: global_BackgroundColor_100.value,
+    display: 'flex',
+    flexDirection: 'column',
+    height: '100%',
+  },
+  topContainer: {
+    justifyContent: 'space-between',
+    padding: '16px 24px',
+    height: 'fit-content',
+  },
+  bottomContainer: {
+    justifyContent: 'space-between',
+    height: 'fit-content',
+  },
+});
+
+const perPageKey = 'packagePerPage';
+
+export default function SnapshotPackageModal() {
+  const { contentOrigin } = useAppContext();
+  const classes = useStyles();
+  const { repoUUID: uuid } = useParams();
+  const rootPath = useRootPath();
+  const navigate = useNavigate();
+  const storedPerPage = Number(localStorage.getItem(perPageKey)) || 20;
+  const [page, setPage] = useState(1);
+  const [perPage, setPerPage] = useState(storedPerPage);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [activeSortIndex, setActiveSortIndex] = useState<number>(0);
+  const [activeSortDirection, setActiveSortDirection] = useState<'asc' | 'desc'>('asc');
+
+  const columnHeaders = ['Name', 'Version', 'Release', 'Arch'];
+
+  const columnSortAttributes = ['name', 'version', 'release', 'arch'];
+
+  const sortString = useMemo(
+    () => columnSortAttributes[activeSortIndex] + ':' + activeSortDirection,
+    [activeSortIndex, activeSortDirection],
+  );
+
+  const debouncedSearchQuery = useDebounce(searchQuery);
+
+  useEffect(() => {
+    setPage(1);
+  }, [debouncedSearchQuery, sortString]);
+
+  const {
+    isLoading,
+    isFetching,
+    isError,
+    data = { data: [], meta: { count: 0, limit: 20, offset: 0 } },
+  } = useGetPackagesQuery(uuid as string, page, perPage, debouncedSearchQuery, sortString);
+
+  useEffect(() => {
+    if (isError) {
+      onClose();
+    }
+  }, [isError]);
+
+  const onSetPage = (_, newPage) => setPage(newPage);
+
+  const onPerPageSelect = (_, newPerPage, newPage) => {
+    setPerPage(newPerPage);
+    setPage(newPage);
+    localStorage.setItem(perPageKey, newPerPage.toString());
+  };
+
+  const sortParams = (columnIndex: number, isDisabled: boolean): ThProps['sort'] | undefined => {
+    if (isDisabled) return;
+    return {
+      sortBy: {
+        index: activeSortIndex,
+        direction: activeSortDirection,
+        defaultDirection: 'asc',
+      },
+      onSort: (_event, index, direction) => {
+        setActiveSortIndex(index);
+        setActiveSortDirection(direction);
+      },
+      columnIndex,
+    };
+  };
+
+  const onClose = () =>
+    navigate(rootPath + (contentOrigin === ContentOrigin.REDHAT ? `?origin=${contentOrigin}` : ''));
+
+  const {
+    data: packageList = [],
+    meta: { count = 0 },
+  } = data;
+
+  const fetchingOrLoading = isFetching || isLoading;
+
+  const loadingOrZeroCount = fetchingOrLoading || !count;
+
+  return (
+    <Modal
+      key={uuid}
+      position='top'
+      hasNoBodyWrapper
+      aria-label='RPM package modal'
```

Comment:
i think you'll want to change this aria-label and ouiaId from RPM package modal to snapshot package modal to differentiate between the 2 modals